external-crates/move/crates/move-compiler/src/parser/ast.rs

```diff
@@ -72,11 +72,22 @@ pub struct Program {
     pub lib_definitions: Vec<PackageDefinition>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
+pub enum PkgDefKind {
+    /// source - full compilation
+    Source,
+    /// library - skip function bodies
+    Library,
+    /// skipped source - skip function bodies
+    Skipped,
```

Comment:
I am a bit confused here. If we have "source - full compilation", how does that end up in `TargetKind::External` which says "A dependency only used for linking. No bytecode or CompiledModules are generated"

Reply:
In short, I'm a bit confused by the two separate enums. I was imagining modifying the previous enum to capture the possible states, but this seems to just layer weirdly, which creates states that can't exist?

Reply:
Makes sense. I re-used (and extended) existing `TargetKind` enum to avoid this